### PR TITLE
activate-unity-license@1.0.1

### PR DESCRIPTION
**Pull request by StephenHodgson**
- fall back to hub licensing client if `UNITY_EDITOR_PATH` not defined

### CODE REVIEW

File: dist/index.js [hunk 1]
```diff
@@ -28704,15 +28700,21 @@ let client = undefined;
...
+    let useLicenseClient = false;
     const editorPath = process.env.UNITY_EDITOR_PATH;
-    const version = process.env.UNITY_EDITOR_VERSION || editorPath.match(/(\d+\.\d+\.\d+[a-z]?\d?)/)[0];
-    core.debug(`Unity Editor Path: ${editorPath}`);
-    core.debug(`Unity Version: ${version}`);
-    await fs.access(editorPath, fs.constants.X_OK);
+    if (editorPath) {
+        core.debug(`Unity Editor Path: ${editorPath}`);
+        await fs.access(editorPath, fs.constants.X_OK);
+        const version = process.env.UNITY_EDITOR_VERSION || editorPath.match(/(\d+\.\d+\.\d+[a-z]?\d?)/)[0];
```
Comment (afarchy) [inline dist/index.js:28708]: I think it's good practice to leave a comment example of what the regex is trying to parse.

// Example: 2020.3.13f1

File: src/licensing-client.js [hunk 2]
```diff
@@ -8,15 +8,21 @@ let client = undefined;
...
-    core.debug(`Unity Editor Path: ${editorPath}`);
-    core.debug(`Unity Version: ${version}`);
-    await fs.access(editorPath, fs.constants.X_OK);
+    if (editorPath) {
+        core.debug(`Unity Editor Path: ${editorPath}`);
+        await fs.access(editorPath, fs.constants.X_OK);
+        const version = process.env.UNITY_EDITOR_VERSION || editorPath.match(/(\d+\.\d+\.\d+[a-z]?\d?)/)[0];
+        core.debug(`Unity Version: ${version}`);
+        const major = version.split('.')[0];
+        // if 2019.3 or older, use unity hub licensing client
```
Comment (afarchy) [inline src/licensing-client.js:19]: // explain why

**Review by afarchy — APPROVED** [review]
looks good, add comments

File: src/licensing-client.js (hunk 2 above)
Comment (StephenHodgson) [reply src/licensing-client.js:19]: the license client doesn't exist in the editor installation prior to 2020

File: dist/index.js (hunk 1 above)
Comment (StephenHodgson) [reply dist/index.js:28708]: regex is parsing version. hence the name of the variable it sets.

**Pull request merged**